components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -199,9 +211,13 @@ public void updateApplication(ServiceProvider serviceProvider, String tenantDoma
 
         // invoking the listeners
         Collection<ApplicationMgtListener> listeners = ApplicationMgtListenerServiceComponent.getApplicationMgtListeners();
-        for (ApplicationMgtListener listener : listeners) {
-            if (listener.isEnable() && !listener.doPreUpdateApplication(serviceProvider, tenantDomain, username)) {
-                return;
+        if (isImportSP.get()) {
+            isImportSP.set(false);
```

Comment:
Don't modify the TL here, instead check if it is set and invoke listeners based on that. Unsetting this should be done at the callee as discussed offline